### PR TITLE
Adds Gas Analyzer

### LINKED ISSUE #1584: Gas analyzer
We need an in-game tool for analyzing the atmosphere.

### PR DESCRIPTION
**Pull request by exp111**
grafik
Closes #1584

### CODE REVIEW

**Comment by RemieRichards** [comment]
Feels a bit gross (ab)using tooltips like this.

**Comment by exp111** [comment]
Floating text was not good enough to read long text and I thought the chat was not "direct" enough

**Comment by RemieRichards** [comment]
I was thinking it probably needs a full window.

I know that's quite a lot of boilerplate for a few lines of info though.

Is there anything cool you could add to a full-scale window? display the icons of the gases? pie chart of the tile's total mix percentages?

File: Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs [hunk 1]
```diff
@@ -1,40 +1,59 @@
...
+            {
+                _notifyManager.PopupMessage(eventArgs.ClickLocation, eventArgs.User, "No atmosphere!");
+                return;
+            }
 
-            message.AddText($"Pressure: {tile.Pressure}\n");
-            message.AddText($"Temperature: {tile.Temperature}\n");
+            string message = "";
+            message += $"Pressure: {tile.Pressure:0.##} kPa\n";
+            message += $"Temperature: {tile.Temperature}K ({TemperatureHelpers.KelvinToCelsius(tile.Temperature)}°C)\n";
```
Comment (Hugal31) [inline Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs:45]: Please use localized strings, string interpolation is not localizable :
```c#
message += Loc.GetString("Pressure: {0.##} kPa", tile.Pressure) + '\n'
```

File: Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs [hunk 2]
```diff
@@ -1,40 +1,59 @@
...
             {
                 var gas = Atmospherics.GetGas(i);
 
                 if (tile.Gases[i] <= Atmospherics.GasMinMoles) continue;
 
-                message.AddText(gas.Name);
-                message.AddText($"\n Moles: {tile.Gases[i]}\n");
+                message += $"{gas.Name}: {tile.Gases[i]} mol \n";
             }
+            message = message.TrimEnd(); // Remove annoying newline
```
Comment (Hugal31) [inline Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs:55]: If you want, you can get rid of this by not adding `\n` for the temperature line, and add the `\n` at the beginning of each gas line.

**Comment by exp111** [comment]
Fixed the reviews, but I think I'm gonna do Remie's idea instead, so this pr will be drafted

File: Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs [hunk 3]
```diff
@@ -1,40 +1,61 @@
...
 
 namespace Content.Server.GameObjects.Components.Atmos
 {
     [RegisterComponent]
-    public class GasAnalyzerComponent : Component, IExamine
+    public class GasAnalyzerComponent : Component, IAfterInteract
     {
+#pragma warning disable 649
+        [Dependency] private IServerNotifyManager _notifyManager = default!;
+        [Dependency] private ILocalizationManager _loc = default!;
```
Comment (RemieRichards) [inline Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs:23]: Don't need to pull this in.

Localisation manager is globally available as `Loc`, in `Robust.Shared.Localization`

File: Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs [hunk 4]
```diff
@@ -19,14 +20,15 @@ public class GasAnalyzerComponent : Component, IAfterInteract
     {
 #pragma warning disable 649
         [Dependency] private IServerNotifyManager _notifyManager = default!;
+        [Dependency] private ILocalizationManager _loc = default!;
```
Comment (Hugal31) [inline Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs:23]: You should be able to use plain `Loc.GetString` (a static convenience wrapper), which will save you this dependency.

**Comment by exp111** [comment]
Current images:
UI:
grafik
Tooltip hover;
grafik
ItemStatus:
grafik

**Comment by Zumorica** [comment]
Pressure units need to be specified in that first image.

**Pull request merged**